src/main/java/hudson/plugins/git/util/BuildData.java

```diff
@@ -243,4 +243,22 @@ public String toString() {
                 ",buildsByBranchName="+buildsByBranchName+
                 ",lastBuild="+lastBuild+"]";
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
So sorry that I dropped this from the earlier revert.  I still don't understand how I made that mistake, but it clearly looks like it was my mistake.  I thought I had used a pure "git revert", yet that's not what it shows.

Doesn't adding an equals method without a hashCode method mean that this object will violate the hashCode contract?  I thought that it was a known findbugs warning if equals() is implemented in a class, but hashCode() is not  implemented.